crates/autopilot/src/domain/settlement/trade.rs

```diff
@@ -44,3 +80,81 @@ pub struct ClearingPrices {
     pub sell: eth::U256,
     pub buy: eth::U256,
 }
+
+fn trade_surplus(
+    kind: order::Side,
+    executed: order::TargetAmount,
+    sell: eth::Asset,
+    buy: eth::Asset,
+    prices: &ClearingPrices,
+) -> Result<eth::Asset, error::Math> {
+    match kind {
```

Comment:
Is there are reason why this is not inlined into `Trade::surplus()`?

Reply:
Yup, mentioned here: https://github.com/cowprotocol/services/pull/2504#discussion_r1540767048